File: rotp-server/src/main/java/com/roa/rotp/common/entity/AuditLog.java
package com.roa.rotp.common.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

import java.util.List;

@Entity
@Getter
@Setter
@Builder
@Table(name = "audit_log")
@AllArgsConstructor
public class AuditLog extends BaseEntity {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String action;
    private String result;
    private String message;

    private String ipAddress;
    private String userAgent;
    private String requestUri;
    private String httpMethod;

    @Column(length = 4000)
    private String requestPayload;

    public static final List<String> SEARCHABLE_FIELDS =
            List.of("userAgent", "ipAddress", "requestUri");

    public AuditLog() {}
}

File: rotp-admin/src/types/auditLog.ts
import type { SearchParams } from "./search";

export interface AuditLogSearchRequest {
    page: number;
    size: number;
    search?: SearchParams;
    action?: string;
    result?: string;
    ipAddress?: string;
    userAgent?: string;
    requestUri?: string;
    httpMethod?: string;
    startDate?: string;
    endDate?: string;
}

export interface AuditLogResponse {
    id: number;
    action: string;
    result: string;
    message: string | null;
    ipAddress: string | null;
    userAgent: string | null;
    requestUri: string | null;
    httpMethod: string | null;
    requestPayload: string | null;
    createdAt: string;
    createdBy: string | null;
}

export const AuditLogSearchableFields = ["userAgent", "ipAddress", "requestUri"];
